app/services/provider_interface/reference_status_report_service.rb

```diff
@@ -0,0 +1,43 @@
+module ProviderInterface
+  class ReferenceStatusReportService
+    attr_reader :provider_application_references, :context
+    include Rails.application.routes.url_helpers
```

Comment:
since we're only calling this once we could move this directly onto the link itself